bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,213 @@
+                logger.debug("Exception during poll", e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        this.config = config;
+
+        if (!config.host.isEmpty() && !config.password.isEmpty()) {
+            host = config.host;
+            password = config.password;
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            logger.debug("Initializing EnergeniePWMHandler for Host '{}'", host);
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergeniePWMHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    public void getState() {
+        String url = "http://" + host + "/login.html";
+        String urlParameters = "pw=" + password;
+        InputStream urlContent = new ByteArrayInputStream(urlParameters.getBytes(StandardCharsets.UTF_8));
+        String loginResponseString = null;
+
+        try {
+            logger.trace("sending 'POST' request to URL : {}", url);
+            loginResponseString = HttpUtil.executeUrl("POST", url, urlContent, "TEXT/PLAIN", HTTP_TIMEOUT_MILLISECONDS);
+
+            if (loginResponseString != null) {
+                updateState("voltage", PWMState.VOLTAGE.readState(loginResponseString));
+                updateState("current", PWMState.CURRENT.readState(loginResponseString));
+                updateState("power", PWMState.POWER.readState(loginResponseString));
+                updateState("energy", PWMState.ENERGY.readState(loginResponseString));
+                try {
+                    HttpUtil.executeUrl("POST", url, HTTP_TIMEOUT_MILLISECONDS);
+                    logger.trace("logout from ip {}", host);
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "failed to logout: " + e.getMessage());
+                }
+            }
+
+        } catch (IOException e) {
+            logger.debug("energenie: failed to login to {} with ip {}", thing.getUID(), host, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private synchronized void onUpdate() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::getState, 5, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
```

Comment:
This will make the warning go away.
```suggestion
    private synchronized void onUpdate() {
        ScheduledFuture<?> refreshJob = this.refreshJob;
        if (refreshJob == null || refreshJob.isCancelled()) {
            this.refreshJob = scheduler.scheduleWithFixedDelay(this::getState, 5, refreshInterval, TimeUnit.SECONDS);
        }
    }
```

Reply:
Done